src/wasm/wasm-binary.cpp

```diff
@@ -1581,6 +1682,18 @@ void WasmBinaryBuilder::readFunctionSignatures() {
   }
 }
 
+Signature WasmBinaryBuilder::getFunctionSignatureByIndex(Index index) {
+  Signature sig;
+  if (index < functionImports.size()) {
+    return functionImports[index]->sig;
+  }
+  Index adjustedIndex = index - functionImports.size();
+  if (adjustedIndex >= functionSignatures.size()) {
+    throwError("invalid call index");
```

Comment:
I know this is pre-existing, but
```suggestion
    throwError("invalid function index");
```